src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -21,58 +22,58 @@ def numberOfRowsInTableView_(self, table) -> int:
         return len(self.interface.data) if self.interface.data else 0
 
     @objc_method
-    def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
-        data_row = self.interface.data[row]
-
-        try:
-            # Obtain the _impl for the data row...
-            data = data_row._impl
-        except AttributeError:
-            # or if it doesn't already exist, create it.
-            data = {}
-            data_row._impl = data
+    def tableView_viewForTableColumn_row_(self, table, column, row: int):
 
+        data_row = self.interface.data[row]
         col_identifier = str(column.identifier)
 
-        try:
-            # Get the TogaData
-            datum = data[col_identifier]
-        except KeyError:
-            # or create it, if it doesn't exist
-            data[col_identifier] = TogaData.alloc().init()
-            data_row._impl[col_identifier] = data[col_identifier]
-            datum = data[col_identifier]
-
-        # Get value for the column
         try:
             value = getattr(data_row, col_identifier)
+
+            # if the value is a widget itself, just draw the widget!
+            if isinstance(value, toga.Widget):
+                return value._impl.native
+
+            # Allow for an (icon, value) tuple as the simple case
+            # for encoding an icon in a table cell. Otherwise, look
+            # for an icon attribute.
+            elif isinstance(value, tuple):
+                icon_iface, value = value
+            else:
+                try:
+                    icon_iface = value.icon
+                except AttributeError:
+                    icon_iface = None
         except AttributeError:
-            # The accessor doesn't exist in the data. Use the missing value.
-            try:
-                value = self.interface.missing_value
-            except ValueError as e:
-                # There is no explicit missing value. Warn the user.
-                message, value = e.args
-                print(message.format(row, col_identifier))
-
-        # Allow for an (icon, value) tuple as the simple case
-        # for encoding an icon in a table cell.
-        if isinstance(value, tuple):
-            icon, value = value
+            # If the node doesn't have a property with the
+            # accessor name, assume an empty string value.
```

Comment:
We recently added "missing value" handling to table; (see lines 51-56 in the old version), rather than using a blank string.

Reply:
Ah, I had overlooked this.